aether-ui/aether/ui/assets/apps/pipeline/Pipeline.jsx

```diff
@@ -23,16 +23,16 @@ import { connect } from 'react-redux'
 import { FormattedMessage } from 'react-intl'
 import { Link } from 'react-router-dom'
 
-import { ModalError, NavBar } from '../components'
+import { ModalError, NavBar, Modal } from '../components'
 
-import ContractAddButton from './components/ContractAddButton'
 import Input from './sections/Input'
 import EntityTypes from './sections/EntityTypes'
 import Mapping from './sections/Mapping'
 import Output from './sections/Output'
 import Settings from './sections/Settings'
 
-import { clearSelection, getPipelineById, selectContract, selectSection } from './redux'
+import { clearSelection, getPipelineById, selectContract, selectSection, contractChanged } from './redux'
```

Comment:
Could you move "contractChanged" & friends to the settings component, this component distribute the actions but does not save anything, keep this file as clean and simple as possible :pray:

Reply:
`contractChanged` is used outside the settings component.

Reply:
That's my point, it should be used only there...

Reply:
It is used for all contract related components like [here0(https://github.com/eHealthAfrica/aether/pull/517/files#diff-7ed5fd717c77490e4e6926a5f0ff1eb7R89) for saving changes to unsaved contracts. This feature is currently silenced by the cancel modal on exiting the settings component. But i think will change down the line.

Reply:
@obdulia-losantos could you please explain more how you would go about doing this?

Reply:
`contractChanged` action has been limited to the settings component and used to set the redux `currentContract` to a new blank contract while `selectContract` is used to set an existing contract with passed ids.